docs/syntax-jsx-support.md

```diff
@@ -0,0 +1,338 @@
+    local className = assert(Intrinsics[tag], ("Unknown intrinsic tag %q"):format(tag))
+
+    local inst = Instance.new(className)
+    props = props or {}
+
+    -- Apply properties
+    for k, v in pairs(props) do
+        inst[k] = v
+    end
+
+    -- Parent children to this instance
+    for _, child in ipairs({ ... }) do
+        if typeof(child) == "Instance" then
+            child.Parent = inst
+        end
+    end
+
+    return inst
+end
+
+local scene =
+    <model Name="Root">
+        <part Name="Ground" Anchored={true} Size={Vector3.new(50, 1, 50)} />
+        <part Name="Lamp" Position={Vector3.new(0, 5, 0)}>
+            <pointlight Brightness={3.5} Range={24} />
+        </part>
+    </model>
+```
+
+This keeps the same core idea: the tag is a string, props are a table, and children are a list. The meaning is defined by the chosen runtime factory (`jsx`), and JSX is simply a concise way to express the tree.
+
+### Compatibility & ambiguity analysis
+
+#### Backwards compatibility
+
+With `.luax` + flag gating, existing `.lua`/`.luau` source is unaffected.
+
+Within `.luax`, code that uses `<` for comparisons (e.g. `a < b`) remains valid; JSX is only recognized when the parser expects a `simpleexp` and sees `<` as the next token in that position.
+
+#### Grammar ambiguity concerns
+
+Key ambiguity sources:
+
+- `<` as binary operator vs JSX element start
+- `</` sequences inside other contexts
+- Interaction with existing type syntax, including explicit type instantiation (`<<T>>()`)
+
+Mitigations:
+
+- Restrict JSX recognition to positions where a `simpleexp` is parsed.
+- Keep JSX syntax small initially (no raw text children, fragments, or spreads).
+- Continue gating behind a flag to allow collecting real-world compatibility data.
+
+### Editor integration & tooling considerations
+
+Adding JSX affects:
+
+- Tokenization/highlighting for `.luax`
+- Parsing for autocomplete, formatter/pretty-printer, and incremental parse
+- Error recovery expectations (JSX introduces new “paired delimiter” structures)
+
+Mitigations:
+
+- Keep JSX AST lowering to existing nodes so downstream tools operate on familiar AST shapes.
+- Ensure error recovery always makes progress to avoid hangs in partial code states.
+- Treat `.luax` as a distinct language mode/extension for editors.
+
+### Security / sandboxing considerations
+
+The syntax itself is not inherently unsafe, but it makes calling a factory function extremely easy. Sandboxing concerns are the same as for ordinary function calls:
+
+- The host controls which globals exist (including `jsx`).
+- Environments can restrict or replace `jsx` to safe implementations.
+
+### Performance considerations
+
+- Parsing adds new branches and lookahead, but is expected to be negligible compared to overall parsing cost.
+- Desugaring into existing AST nodes avoids adding new runtime node kinds and keeps later stages (typechecking, codegen) unchanged.
```

Comment:
Luau's code base works directly on a single AST today. You're proposing adding a multi-stage compiler effectively, which while reasonable is, again, a much more significant architectural change to the product than how you're trying to describe it here.